compiler/src/dotty/tools/dotc/core/SymDenotations.scala

```diff
@@ -1611,7 +1611,7 @@ object SymDenotations {
                   case _ => false
 
         if owner.isClass then
-          for c <- owner.info.decls.toList if maybeChild(c) do
+          for c <- owner.info.decls.toList if maybeChild(c) && (c.isClass || c.is(EnumVal)) do
```

Comment:
This duplicates similar checks done in `maybeChild`, this can probably be avoided by tweaking maybeChild a bit.